### ISSUE TITLE
Unable to install upgrade-assistant

### ISSUE THREAD

**Issue by Manookian** [dotnet/upgrade-assistant#1137]
Problem encountered on https://dotnet.microsoft.com/en-us/platform/upgrade-assistant/tutorial/install-upgrade-assistant
Operating System: windows

Provide details about the problem you're experiencing. Include your operating system version, exact error message, code sample, and anything else that is relevant.

C:\Users\rxmanook\AppData\Local\Temp\0njm4kfv.0we\restore.csproj : error NU1301: Failed to retrieve information about '
upgrade-assistant' from remote source 'https://lsm-na.pkgs.visualstudio.com/_packaging/281ce2c6-41be-41e9-9e45-aed68ad0
e1f0/nuget/v3/flat2/upgrade-assistant/index.json'.
The tool package could not be restored.
Tool 'upgrade-assistant' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool

**Comment by twsouthwick** [comment]
@Manookian  are you trying to install it from a directory with custom NuGet sources? Please ensure you're doing it in a non-project folder

**Comment by twsouthwick** [comment]
You probably want to try this command:

```
dotnet tool install -g upgrade-assistant --ignore-failed-sources
```

**Comment by Manookian** [comment]
We have a “custom” nuget repo, how do I get it to stop looking there?

**Comment by twsouthwick** [comment]
Run with the `--ignore-failed-sources` command.  If you have NuGet.org disabled, you'll have to either enable it or add the package to your internal feed

**Comment by Manookian** [comment]
Thanks for your help but all has been unsuccessful, not sure how to determine if "NuGet.org disabled"? I have no problems adding nuget packages to my projects so I assume that NuGet.org is enabled as most of the packages we use are NOT in our custom repo but publicly available packages.

**Comment by twsouthwick** [comment]
What's the result if you run with `--ignore-failed-sources`?

This error should not be seen with that:

```
C:\Users\rxmanook\AppData\Local\Temp\0njm4kfv.0we\restore.csproj : error NU1301: Failed to retrieve information about '
upgrade-assistant' from remote source 'https://lsm-na.pkgs.visualstudio.com/_packaging/281ce2c6-41be-41e9-9e45-aed68ad0
e1f0/nuget/v3/flat2/upgrade-assistant/index.json'.
```

Have you tried doing it in a non-project directory, such as `C:\`?

**Comment by Manookian** [comment]
From C:\ here's the error, why is it looking in my local user folders?
C:\Users\rxmanook\AppData\Local\Temp\ty2nui4g.l3g\restore.csproj : error NU1100: Unable to resolve 'upgrade-assistant (
>= 0.3.326103)' for 'net6.0'.
C:\Users\rxmanook\AppData\Local\Temp\ty2nui4g.l3g\restore.csproj : error NU1100: Unable to resolve 'upgrade-assistant (
>= 0.3.326103)' for 'net6.0/any'.
The tool package could not be restored.
Tool 'upgrade-assistant' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool

**Comment by Manookian** [comment]
Command:
dotnet tool install upgrade-assistant --global --version 0.3.326103 --ignore-failed-sources

**Comment by twsouthwick** [comment]
Have you tried the troubleshooting at https://aka.ms/failure-installing-tool? Specifically this one would be a good one to start with:

```
dotnet tool install -g --add-source 'https://api.nuget.org/v3/index.json' --ignore-failed-sources upgrade-assistant
```

**Comment by Manookian** [comment]
Success, added Nuget.org, https://api.nuget.org/v3/index.json to my Package sources in VS. Not sure why this is not default in? Once again, that you for your assistance.

**Comment by twsouthwick** [comment]
It usually is by default - something you did (or your organization) may have have removed it. Glad to hear it worked!

**Comment by nickdevore** [comment]
> It usually is by default - something you did (or your organization) may have have removed it. Glad to hear it worked!

I was having this problem today.  I took what @twsouthwick said to heart and looked in "other" places that visual studio looks for nuget.config files.  In VS2022 (v17.10.1) I went Tools -> Options -> Nuget Package Manager -> Configuration Files and noticed one was at

`C:\Users\{Me}\AppData\Roaming\NuGet\NuGet.Config`

..and sure enough, in there was a local feed I'd created a long time ago that was mucking things up.  I removed that feed from the config file, saved it, closed Visual Studio, and then it worked fine from the command prompt.
